pkg/schema/error.go

```diff
@@ -137,6 +144,23 @@ func (t unexpectedKeyError) Error() string {
 	return msg
 }
 
+type invalidSchemaTypeAnnotationError struct {
```

Comment:
```suggestion
type invalidTypeAnnotationError struct {
```

For brevity, can be shortened given that we're in the `schema` package.

Reply:
More fundamental question: whether we need a new error type here.

Is there a significant difference between an `invalidSchemaError` and a `invalidTypeAnnotationError`?  I personally haven't thought deeply about the `schema` package's errors.

I _think_:
- keep the number of `error` types to as few as possible (managing complexity)
- without losing meaningful distinctions (e.g. different data captured in each situation; different error message format desired; potentially different handling by callers...)